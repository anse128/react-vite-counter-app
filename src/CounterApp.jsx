import PropTypes from 'prop-types'

const handleAdd = (event, newValue) => {
    console.log(event)
    // console.log(newValue)
}

export const CounterApp = ({ value }) => {

    // function handleAdd(event, newValue) {
    //     console.log(event)
    //     console.log(newValue)
    // }
    return (
        <>
            <h1>CounterApp</h1>
            <h1>{value}</h1>
            {/* <button onClick={(event) => handleAdd(event, 'Hola')}> */}
            <button onClick={handleAdd}>
                +1
            </button >
        </>

    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
CounterApp.defaultProps = {
    value: 1
}







