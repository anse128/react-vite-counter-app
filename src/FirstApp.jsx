
import PropTypes from 'prop-types' // para definirle el tipo a las propiedades

export const FirstApp = ({ title, subTitle, name }) => { // 


    return (

        <>
            <h1>{title}</h1>
            <p>{subTitle}</p> {/*  */}
            <p>{name}</p>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired, // manda error si no es mandado
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    // title: 'No hay título',
    subTitle: 'No hay subtítulo',
    name: 'hola hp'
}











// // import {Fragment} from 'react'

// // cuando vamos a tener dos elementos en el functionalcomponent no se debe contener en un div
// // para eso se usa un fragment pero se debe de importar
// // lo mejor es usar <> y asi no hay que importar nada

// // const newMessage = 'Fernando' // Regla si la variable no esta cambiando de valor debe de ir fuera del functional component, porque react no va a reprocesar algo que este fuera del componente

// import PropTypes from 'prop-types' // para definirle el tipo a las propiedades

// // const newMessage = {
// //     message: 'Hola Mundo',
// //     title: 'Sebastian'
// // }

// // const getResult = (i) => {
// //     return (2 + 2 + i)
// // }


// // export const FirstApp = (props) => { // no es lo comun usar las props así 
// // export const FirstApp = ({ title = 'Hola, Soy Sebastián'}) => { // valor por defecto
// export const FirstApp = ({ title, subTitle, name }) => { // 

//     // console.log(title)

//     // if (!title) {
//     //     throw new Error('El title no existe')
//     // }


//     return (
//         // <div>
//         //     <h2>FirstApp</h2>
//         //     <h2>FirstApp</h2>
//         // </div>
//         // <Fragment>
//         //     <h2>FirstApp</h2>
//         //     <h2>FirstApp</h2>
//         // </Fragment>
//         <>
//             {/* <h2>{newMessage.title}</h2> */}
//             {/* <h2>{props.title}</h2> */}
//             <h1>{title}</h1>

//             {/* <h2>{getResult(2)}</h2>  no puede ser una función asincrona */}

//             {/* <code>{JSON.stringify(newMessage)}</code> */}

//             <p>{subTitle}</p> {/*  */}
//             <p>{name}</p>
//         </>
//     )
// }


// FirstApp.propTypes = {
//     title: PropTypes.string.isRequired, // manda error si no es mandado
//     subTitle: PropTypes.string
// }

// FirstApp.defaultProps = {
//     title: 'No hay título',
//     subTitle: 'No hay subtítulo',
//     name: 'hola hp'
// }
