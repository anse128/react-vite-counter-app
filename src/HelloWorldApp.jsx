// function App() {     // todos los nombres de sus functional components van capitalizados con PascalCase o UpperCamelCase
//     // document.createElement....
//     return <h1>Hola Mundo!!!</h1>
// }


// export default App

export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  )
}

// export const HelloWorldApp = () => <h1>HelloWorldApp</h1>
