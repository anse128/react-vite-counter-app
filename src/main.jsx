import React from 'react' // imr
import ReactDOM from 'react-dom/client' //imp
import { HelloWorldApp } from './HelloWorldApp'
import { FirstApp } from './FirstApp'

import { CounterApp } from './CounterApp'
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render( // id = root del html
    <React.StrictMode> 
        <CounterApp value={10} />
        {/* <FirstApp  title = "Hola, Soy Vegeta"/> */}
    </React.StrictMode>
)



// react testing library es una libreria que est치 mas enfocada en lo que sucede en la pantalla despues de hacer las interacciones, hacer todo lo que hariamos de manera manual, clicks etc...

// Jest est치 mas orientada a hacer las acerciones y hacer mocks de funciones que queremos evaluar












// import React from 'react' // imr
// import ReactDOM from 'react-dom/client' //imp
// import { HelloWorldApp } from './HelloWorldApp'
// import { FirstApp } from './FirstApp'

// import { CounterApp } from './CounterApp'
// import './styles.css'
// // componente inicial
// //los componentes no suelen estar en el main
// // function App() {     // todos los nombres de sus functional components van capitalizados con PascalCase o UpperCamelCase
// //     // document.createElement....
// //     return <h1>Hola Mundo!!!</h1>
// // }

// // forma de renderizar el componente
// ReactDOM.createRoot(document.getElementById('root')).render( // id = root del html
//     <React.StrictMode> {/* se aconseja trabajar en modo estricto*/}
//         {/* <HelloWorldApp />        renderizamos el app */}
//         {/* <FirstApp title="Hola, soy Sebastian" subTitle={123} /> */}
//         {/* <FirstApp title /> booleano */}
//         <FirstApp />
//         {/* <CounterApp value={10} /> */}
//         {/* <FirstApp  /> */}
//     </React.StrictMode>
// )



// // react testing library es una libreria que est치 mas enfocada en lo que sucede en la pantalla despues de hacer las interacciones, hacer todo lo que hariamos de manera manual, clicks etc...

// // Jest est치 mas orientada a hacer las acerciones y hacer mocks de funciones que queremos evaluar



