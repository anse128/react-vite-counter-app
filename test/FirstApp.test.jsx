import {  getAllByText, render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {

    // test('Debe de hacer match con el snapshot', () => {

    //     const title = 'Hola, Soy Goku'
    //     const { container } = render(<FirstApp title={title} />)

    //     expect(container).toMatchSnapshot()
    // })

    test('Debe de mostrar el tÃ­tulo en un h1', () => {
        const title = 'Hola, Soy Goku'
        const { container, getByText, getByTestId } = render(<FirstApp title={title} />)

        // expect(getByText(title)).toBeTruthy()

        // const h1 = container.querySelector('h1')
        // expect(h1.innerHTML).toBe(title)
        // expect(h1.innerHTML).toContain(title) //No importa si hay espacios antes o despues

        expect(getByTestId('test-title').innerHTML).toContain(title)
    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const title = 'Hola, Soy Goku'
        const subTitle = 'Soy un subtitulo'

        const { getAllByText } = render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        )

        // expect(getByText(subTitle )).toBeTruthy()  // se asegura que solo sea uno y da error si hay mas de una
        expect(getAllByText(subTitle).length).toBe(2)
        expect(getAllByText(subTitle).length).toBe(2)
        expect(getAllByText(subTitle).length).toBe(2)
    })



})