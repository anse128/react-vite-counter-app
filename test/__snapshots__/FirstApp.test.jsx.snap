// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
   
  <p>
    hola hp
  </p>
</div>
`;
